iothub.class.nut

```diff
@@ -5,724 +5,1024 @@
-    }
+        static function encodeUriComponentStrict(str) {
+            // NOTE: This may not encode enough characters. If it is a problem, check
+            //       it is encoding: !, ", *, ( and ).
+            return http.urlencode({s=str}).slice(2);
+        }
 
-    static function versionQueryString() {
-        return "?api-version=2016-02-03";
+        static function stringToSign(resourceUri, expiry) {
+            return resourceUri + "\n" + expiry;
+        }
+
+        static function hmacHash(password, stringToSign) {
+            local decodedPassword = http.base64decode(password);
+            local hmac = http.hash.hmacsha256(stringToSign, decodedPassword);
+            return http.base64encode(hmac);
+        }
     }
-}
 
-//------------------------------------------------------------------------------
-class iothub.Authorization {
+    SharedAccessSignature = class {
 
+        sr = null;
+        sig = null;
+        skn = null;
+        se = null;
 
-    static function anHourFromNow() {
-        return time() + 3600;
-    }
+        static function create(resourceUri, keyName, key, expiry) {
 
-    static function encodeUriComponentStrict(str) {
-        // NOTE: This may not encode enough characters. If it is a problem, check
-        //       it is encoding: !, ", *, ( and ).
-        return http.urlencode({s=str}).slice(2);
-    }
+            // The create method shall create a new instance of SharedAccessSignature with properties: sr, sig, se, and optionally skn.
+            local sas = iothub.SharedAccessSignature();
 
-    static function stringToSign(resourceUri, expiry) {
-        return resourceUri + "\n" + expiry;
-    }
+            // The sr property shall have the value of resourceUri.
+            sas.sr = resourceUri;
 
-    static function hmacHash(password, stringToSign) {
-        local decodedPassword = http.base64decode(password);
-        local hmac = http.hash.hmacsha256(stringToSign, decodedPassword);
-        return http.base64encode(hmac);
-    }
-}
+            // <signature> shall be an HMAC-SHA256 hash of the value <stringToSign>, which is then base64-encoded.
+            // <stringToSign> shall be a concatenation of resourceUri + "\n" + expiry.
+            local hash = iothub.Authorization.hmacHash(key, iothub.Authorization.stringToSign(resourceUri, expiry));
+
+            // The sig property shall be the result of URL-encoding the value <signature>.
+            sas.sig = iothub.Authorization.encodeUriComponentStrict(hash);
 
-//------------------------------------------------------------------------------
-class iothub.SharedAccessSignature {
+            // If the keyName argument to the create method was falsy, skn shall not be defined.
+            // <urlEncodedKeyName> shall be the URL-encoded value of keyName.
+            // The skn property shall be the value <urlEncodedKeyName>.
+            if (keyName) sas.skn = iothub.Authorization.encodeUriComponentStrict(keyName);
```

Comment:
This function name was in the original library as well. Seems to run without issue.